src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1129,6 +1129,45 @@ private Conversion ClassifyImplicitBuiltInConversionFromExpression(BoundExpressi
             return Conversion.NoConversion;
         }
 
+#nullable enable
+        private Conversion GetImplicitCollectionExpressionConversion(BoundUnconvertedCollectionExpression collectionExpression, TypeSymbol destination, CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            var collectionExpressionConversion = GetCollectionExpressionConversion(collectionExpression, destination, ref useSiteInfo);
+            if (collectionExpressionConversion.Exists)
+            {
+                return collectionExpressionConversion;
+            }
+
+            // strip nullable from the destination
+            //
+            // the following should work and it is an ImplicitNullable conversion
+            //    ImmutableArray<int>? x = [1, 2];
+            if (IsSystemNullable(destination, out var underlyingDestination))
+            {
+                var underlyingConversion = GetCollectionExpressionConversion(collectionExpression, underlyingDestination, ref useSiteInfo);
+                if (underlyingConversion.Exists)
+                {
+                    return new Conversion(ConversionKind.ImplicitNullable, ImmutableArray.Create(underlyingConversion));
+                }
+            }
+
+            return Conversion.NoConversion;
+        }
+
+        private static bool IsSystemNullable(TypeSymbol type, [NotNullWhen(true)] out TypeSymbol? underlyingType)
+        {
+            if (type is NamedTypeSymbol nt
+                && nt.OriginalDefinition.GetSpecialTypeSafe() == SpecialType.System_Nullable_T)
```

Comment:
It looks like we can use the `SpecialType` property directly. Consider combining all and using a pattern:
```
type is NamedTypeSymbol
{
    OriginalDefinition.SpecialType: SpecialType.System_Nullable_T,
    TypeArgumentsWithAnnotationsNoUseSiteDiagnostics: [var typeArg]
}
```

Reply:
Or replace this method with calls to the existing helper methods (to avoid adding similar methods):
```
if (destination.IsNullableType())
{
    var underlyingConversion = GetCollectionExpressionConversion(collectionExpression, destination.GetNullableUnderlyingType().Type, ref useSite);
    ...
}
```

Reply:
Actually, given that this method is used for several existing callers that were using `GetSpecialTypeSafe()`, perhaps it makes sense to keep as `GetSpecialTypeSafe()` to reduce risk.